tests/integrationtests/tst_perfparser.cpp

```diff
@@ -0,0 +1,332 @@
+        QVERIFY2(topDownCycleCost != topDownInstructionCost, "Top-Down Cycle Cost should not be equal to Top-Down Instruction Cost");
+    }
+
+    void cleanup()
+    {
+        // Clean up
+        QFile outputFile(m_outputPath);
+        outputFile.remove();
+    }
+
+private:
+    SummaryData m_summaryData;
+    Data::BottomUpResults m_bottomUpData;
+    Data::TopDownResults m_topDownData;
+    Data::CallerCalleeResults m_callerCalleeData;
+    QString m_perfCommand;
+    QString m_outputPath;
+
+    void perfRecord(const QStringList &perfOptions, const QString& exePath, const QStringList &exeOptions)
+    {
+        PerfRecord perf(this);
+        QSignalSpy recordingFinishedSpy(&perf, &PerfRecord::recordingFinished);
+        QSignalSpy recordingFailedSpy(&perf, &PerfRecord::recordingFailed);
+
+        perf.record(perfOptions, m_outputPath, exePath, exeOptions);
+
+        QVERIFY(recordingFinishedSpy.wait(10000));
+
+        QCOMPARE(recordingFailedSpy.count(), 0);
+        QCOMPARE(recordingFinishedSpy.count(), 1);
+        QCOMPARE(QFileInfo::exists(m_outputPath), true);
+
+        m_perfCommand = perf.perfCommand();
+    }
+
+    void testPerfData(const Data::Symbol &topBottomUpSymbol, const Data::Symbol &topTopDownSymbol)
+    {
+        PerfParser parser(this);
+
+        QSignalSpy parsingFinishedSpy(&parser, &PerfParser::parsingFinished);
+        QSignalSpy parsingFailedSpy(&parser, &PerfParser::parsingFailed);
+        QSignalSpy summaryDataSpy(&parser, &PerfParser::summaryDataAvailable);
+        QSignalSpy bottomUpDataSpy(&parser, &PerfParser::bottomUpDataAvailable);
+        QSignalSpy topDownDataSpy(&parser, &PerfParser::topDownDataAvailable);
+        QSignalSpy callerCalleeDataSpy(&parser, &PerfParser::callerCalleeDataAvailable);
+
+        parser.startParseFile(m_outputPath, "", "", "", "", "", "");
+
+        QVERIFY(parsingFinishedSpy.wait(6000));
+
+        // Verify that the test passed
+        QCOMPARE(parsingFailedSpy.count(), 0);
+        QCOMPARE(parsingFinishedSpy.count(), 1);
+
+        // Verify the summary data isn't empty
+        QCOMPARE(summaryDataSpy.count(), 1);
+        QList<QVariant> summaryDataArgs = summaryDataSpy.takeFirst();
+        m_summaryData = qvariant_cast<SummaryData>(summaryDataArgs.at(0));
+        QCOMPARE(m_perfCommand, m_summaryData.command);
+        QVERIFY(m_summaryData.sampleCount > 0);
+        QVERIFY(m_summaryData.applicationRunningTime > 0);
+        QVERIFY(m_summaryData.cpusAvailable > 0);
+        QVERIFY(m_summaryData.processCount > 0);
+        QCOMPARE(m_summaryData.cpuArchitecture, QSysInfo::currentCpuArchitecture());
+        QCOMPARE(m_summaryData.linuxKernelVersion, QSysInfo::kernelVersion());
+        QCOMPARE(m_summaryData.hostName, QSysInfo::machineHostName());
+
+        // Verify the sample frequency is acceptable, greater than 500Hz
+        double frequency = m_summaryData.sampleCount / (double)(m_summaryData.applicationRunningTime / (double)1000000000);
```

Comment:
you could rewrite this as:

double frequency = (1E9 * m_summaryData.sampleCount) / m_summaryData.applicationRunningTime;